craft_parts/executor/part_handler.py

```diff
@@ -388,8 +389,51 @@ def _run_stage(
             files=contents.files,
             directories=contents.dirs,
             overlay_hash=overlay_hash.hex(),
+            backstage_files=backstage_contents.files,
+            backstage_directories=backstage_contents.dirs,
         )
 
+    def _migrate_to_backstage(self) -> StepContents:
+        """Migrate files from the part's buildout directory to the backstage directory."""
+        migratable = set(self._part.part_buildout_dir.rglob("*"))
+        migratable_dirs = {
+            str(path.relative_to(self._part.part_buildout_dir))
+            for path in migratable if path.is_dir()
+        }
+        migratable_files = {
+            str(path.relative_to(self._part.part_buildout_dir))
+            for path in migratable if path.is_file()
+        }
+
+        def pkgconfig_fixup(
+            file_path: str,
+            prefix_prepend: Path = self._part.stage_dir,
+            prefix_trim: Path = self._part.part_install_dir,
+        ) -> None:
+            if os.path.islink(file_path):
+                return
+            if not file_path.endswith(".pc"):
+                return
+            packages.fix_pkg_config(
+                prefix_prepend=self._part.backstage_dir,
+                pkg_config_file=Path(file_path),
+                prefix_trim=self._part.part_buildout_dir,
+            )
+            packages.fix_pkg_config(
+                prefix_prepend=self._part.backstage_dir,
+                pkg_config_file=Path(file_path),
+                prefix_trim=self._part.part_install_dir,
+            )
```

Comment:
do we have a use case now for this?

Reply:
Not directly - I'm just trying to ensure that the migration to stage and backstage works the same way so that when we do have uses for it in plugins we don't have weird issues related to plugins using backstage.

Reply:
Removed